### ISSUE TITLE
VFS Windows: Display the sharing state and lock state in the 'Status' column of Windows Explorer

### ISSUE THREAD

**Issue by allexzander** [nextcloud/desktop#4854]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you want to have the same feature implemented.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Feature description
It is possible to display at least one icon in the `Status` field for VFS files/folders in Windows Explorer. We can use this to present the sharing state or lock state for items. It should also be possible to display custom icons if we embed them as resources into nextcloud.exe.
The status icon can then be changed dynamically.

### Below is an example of the icon displayed for OneDrive
![image-65](https://user-images.githubusercontent.com/12224014/185061067-5c84580d-ce66-4f67-9239-5f40638f5f66.png)

**Comment by allexzander** [comment]
@tobiasKaminsky

**Comment by allexzander** [comment]
@jancborchardt Which icon should we use for sharing state? Should we use this one from the server https://github.com/nextcloud/server/blob/master/core/img/actions/shared.svg? Should we display an icon for the incoming share too, or, only for the file that we've shared? Should it be the same icon?

What about the lock state? Which icon needs to be used? Is the one we use on the server (solid black padlock) good enough?

**Comment by jancborchardt** [comment]
> Which icon should we use for sharing state? Should we use this one from the server https://github.com/nextcloud/server/blob/master/core/img/actions/shared.svg? Should we display an icon for the incoming share too, or, only for the file that we've shared? Should it be the same icon?

Yes that icon is good, and yes for any type of shares. :) And yes, whichever the type of share, they can all have the same icon.

> What about the lock state? Which icon needs to be used? Is the one we use on the server (solid black padlock) good enough?

Yes! :)
